tool/tsh/tsh_test.go

```diff
@@ -2175,7 +2177,40 @@ func TestSerializeProfilesNoActive(t *testing.T) {
 	}
 	`
 	testSerialization(t, expected, func(f string) (string, error) {
-		return serializeProfiles(nil, nil, f)
+		return serializeProfiles(nil, nil, nil, f)
+	})
+}
+
+func TestSerializeProfilesWithEnvVars(t *testing.T) {
```

Comment:
Thanks for adding the tests, it's a shame that we don't have an established way to test the output of `tsh status` itself. It'd definitely be useful to have some test cases for the behavior implemented in this PR.